manager.py

```diff
@@ -1,398 +1,442 @@
-# -*- coding: utf-8 -*-
```

Comment:
Why is all of `manager.py` marked as changed? Some ideas:
* text encoding should be UTF-8
* line breaks should be line feed only, not carriage return + line feed (the former being the unix standard, the latter being Windows standard, but your text editor should be able to handle either)

Unless there was a style inconsistency previously, please revert it, it makes it impossible to see the diff.